src/App.tsx

```diff
@@ -1,39 +1,65 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component {
+  timerId = 0;
+
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: false });
+    });
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+      this.setState({ hasClock: false });
+    });
 
-        {' time is '}
+    document.removeEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
```

Comment:
Add remove coorectly listener
```suggestion
    document.removeEventListener('click', this.showClock);
```